### ISSUE TITLE
Add more advanced searching capabilities

### ISSUE THREAD

**Issue by mscoutermarsh** [coderintherye/Kivabeta#4]
Advanced search that makes use of API search capabilities.

http://build.kiva.org/api#GET*|loans|search

**Advanced search fields:**
- partner
- status
- gender

**Sorting:**
- popularity
- loan_amount
- oldest
- expiration
- newest
- amount_remaining
- repayment_term

**Comment by mscoutermarsh** [comment]
this is going to be dependent on issue 3. 

https://github.com/coderintherye/Kivabeta/issues/3

**Comment by coderintherye** [comment]
Yeah once we have #3 in this should be pretty straight-forward as a wraparound. There's PHP and Ruby versions of this already available that I can point people to, not sure that we have a JS version anywhere though, so we'll still need this to happen.

**Comment by mscoutermarsh** [comment]
Could you point to the PHP & Ruby versions?

That would be helpful in getting the JS version started.

**Comment by mscoutermarsh** [comment]
@coderintherye I have a couple questions about the API...

 http://build.kiva.org/api#GET*|loans|search

> q(string)
> A query string against which results should be matched.

What's this mean? Can I specify a name, country, activity...etc... to search by? If so, what's the format? There isn't anything in the docs that I could find.

**Comment by coderintherye** [comment]
@mscoutermarsh Yep, so for instance: http://api.kivaws.org/v1/loans/search.json?country_code=us

Not everything is implemented though. I have some code in PHP for doing country to country_code conversion but there is probably a js library somewhere that does that. If you find something not implemented that we need, let me know so I can see about getting it into the API.

**Comment by coderintherye** [comment]
Alright, so long time since I updated, what basically happened is another engineer here started developing a javascript front end for our API and is going to release it as open source next week. I'll update this issue when that happens, and then any work on this project will be to use that project's javascript code to build a flexible frontend that people can easily customize. Sorry for lack of updates for so long.
